File: ios/MRT/DesignSystem/Components/GHStatusDot.swift
import SwiftUI

struct GHStatusDot: View {
    enum Status {
        case online
        case pending
        case error
        case offline
    }

    let status: Status

    var body: some View {
        Circle()
            .fill(color)
            .frame(width: 8, height: 8)
    }

    private var color: Color {
        switch status {
        case .online:
            return GHColors.accentGreen
        case .pending:
            return GHColors.accentYellow
        case .error:
            return GHColors.accentRed
        case .offline:
            return GHColors.textTertiary
        }
    }
}

#Preview {
    ZStack {
        GHColors.bgPrimary.ignoresSafeArea()
        HStack {
            GHStatusDot(status: .online)
            GHStatusDot(status: .pending)
            GHStatusDot(status: .error)
            GHStatusDot(status: .offline)
        }
    }
}

File: android/app/src/main/java/com/mrt/app/designsystem/components/GHStatusDot.kt
package com.mrt.app.designsystem.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.mrt.app.designsystem.theme.GHColors

enum class GHStatus {
    Online,
    Pending,
    Error,
    Offline,
}

@Composable
fun GHStatusDot(
    status: GHStatus,
    modifier: Modifier = Modifier,
) {
    val color = when (status) {
        GHStatus.Online -> GHColors.AccentGreen
        GHStatus.Pending -> GHColors.AccentYellow
        GHStatus.Error -> GHColors.AccentRed
        GHStatus.Offline -> GHColors.TextTertiary
    }

    Box(
        modifier = modifier
            .size(8.dp)
            .clip(CircleShape)
            .background(color),
    )
}
